### PR TITLE
Added feature to view file information in ServerFileFragment

### PR DESCRIPTION
**Pull request by mohitsvnit**
This PR fixes issue #342.

I added an icon to view meta data of file in application tool bar in `ServerFileFragment`.
I used Material Design dialog. We can add more information to display when meta data of file will increase.

screenshot_1521917481

screenshot_1521917471

### CODE REVIEW

File: src/main/res/values/strings.xml [hunk 1]
```diff
@@ -150,4 +151,11 @@
     <string name="intro_title_6">Ready</string>
     <string name="intro_desc_6">You\'re all set to go. Thanks for using Amahi.</string>
 
+    <!--Dialog File Info-->
+    <string name="dialog_file_name">Name:</string>
+    <string name="dialog_file_type">File Type:</string>
+    <string name="dialog_file_modification">Last Modification:</string>
```
Comment (adi23arora) [inline src/main/res/values/strings.xml]: I guess it should be "Last Modified" or just "Modified".

Comment (mohitsvnit) [reply src/main/res/values/strings.xml]: oops! My bad.

File: src/main/res/values/strings.xml [hunk 2]
```diff
@@ -42,6 +42,7 @@
     <string name="menu_settings">Settings</string>
     <string name="menu_share">Share</string>
     <string name="menu_delete">Delete</string>
+    <string name="menu_information">Information</string>
```
Comment (adi23arora) [inline src/main/res/values/strings.xml]: "Properties" would be an apt word for this menu option as we are displaying file properties.

File: src/main/java/org/amahi/anywhere/fragment/ServerFilesFragment.java [hunk 3]
```diff
@@ -339,6 +349,39 @@ public void onClick(DialogInterface dialog, int which) {
...
+                    actionMode.finish();
+
+                }
+            }
+        });
+
+        dialofFileInfo.show();
+    }
+
+    private String getLocalDateTime(Date modificationTime) {;
```
Comment (adi23arora) [inline src/main/java/org/amahi/anywhere/fragment/ServerFilesFragment.java]: Spare semicolon `;`

File: src/main/res/values/strings.xml (hunk 2 above)
Comment (mohitsvnit) [reply src/main/res/values/strings.xml]: I checked it with other apps. They are using "Information". Lets mentor give review on this one.

**Comment by cpg** [comment]
Looks like there are two columns. In terms of usability, if the labels on the left column can be right-aligned, it will look better. In addition, it could be improved if the color of the labels on the left column could be light gray and the colon could be removed, improving things.

**Comment by mohitsvnit** [comment]
Sure will do that.

**Comment by mohitsvnit** [comment]
Color: material grey 500
screenshot from 2018-04-02 14-20-55

Color: material grey 600
screenshot from 2018-04-02 14-20-27


@cpg please review above design

**Comment by cpg** [comment]
Also (sorry, I just realized this) ... is the cancel button really necessary? It takes a lot of space.

Wouldn't simply touching the other area be enough the close this area? Or even also touching this area in the message?

Also, if not possible, how about calling it "Close" or just an `X` somewhere or something (so long as it complies with MD).

I can't tell much about the difference other than the edges (the second one seems better in that respect).

**Comment by mohitsvnit** [comment]
If the user will touch on another area then the dialog will close but the file will still be in the selected mode and user can select another file by just touching it.

Whereas if the user clicks on `CANCEL` button then the file will be deselected. 

For the button, I used predefined button theme. 
I will try to make it more user-friendly.

**Comment by cpg** [comment]
What is the significance of the file being selected or deselected? (seems like not a big deal to me)

Also, what is the user "Cancelling" when touching "cancel"?

**Comment by mohitsvnit** [comment]
You are right. I will remove cancel button.

**Comment by ArihantJain456** [comment]
@mohitsvnit I think the proper alignment of labels is missing.
you can try something like this 
file dialog

**Comment by mohitsvnit** [comment]
screenshot from 2018-04-03 14-06-41

screenshot from 2018-04-03 14-07-08

screenshot from 2018-04-03 14-13-37

@cpg which one do you suggest?

**Comment by cpg** [comment]
definitely first one of the three, in my view.

i would also take away the file type line. who cares about it?
user probably has a good idea of what that file is anyway.

basically, i don't think this feature very important (not high priority), but if we do it, we should do it cleanly and without junk on it.

**Comment by mohitsvnit** [comment]
Ok, I will remove file type. Do you want me to increase text size to `16sp` as suggested in material design guideline? Currently, it is `14sp`. 

The 3rd one is with `16sp` text size.

**Comment by cpg** [comment]
i don't know. it's going to make some file names overflow. I would not mind if it's smaller.

actually, haha ...  honestly, ... without the file info (which is not really necessary), all that information is available in the entry for the file already, so this does not add much value, i think ...

**Comment by mohitsvnit** [comment]
screenshot_1522755242

I updated the code as per everyone suggestions.

**Comment by mohitsvnit** [comment]
screenshot_1522765963

Updated code as per everyone suggestions.

**Comment by cpg** [comment]
OK. After more understanding of what this is an after a design analysis, I'm reticent to push for it. It's not really immediately useful to the user as it is.

The only way I can think this could be useful is if we move things from the nav header to this pop up. For example if we move the "delete the file" button, or the "share the file" button there. That would allow us to add more things over time, at the expense of one extra tap for each such thing

However, the thing is we _do not really need expansion yet._ So I would like to close this now but we can re-open it an use the work done once we need the expansion. Thanks for your patience with my questions an slow response @mohitsvnit 

Closes #342

**Pull request closed without merging**